packages/web/src/components/buy-audio-modal/components/AudioAmountPicker.tsx

```diff
@@ -0,0 +1,142 @@
+          <div className={styles.amountPreviewBuy}>{messages.buy}</div>
+          <div className={styles.amountPreview}>
+            <IconAUDIO />
+            {amount} $AUDIO
+          </div>
+        </>
+      ) : (
+        messages.selectAnAmount
+      )}
+    </div>
+  )
+}
+
+export const AudioAmountPicker = ({
+  presetAmounts,
+  hideCustomAmount,
+  onAmountChanged
+}: {
+  presetAmounts: string[]
+  hideCustomAmount?: boolean
+  onAmountChanged: (amount: string) => void
+}) => {
+  const [isCustomAmountInputVisible, setIsCustomAmountInputVisible] =
+    useState(false)
+  const [value, setValue] = useState<string | null>(null)
+  const [presetAmount, setPresetAmount] = useState<string>()
+  const [customAmount, setCustomAmount] = useState<string>()
+  const handleChange = useCallback(
+    (e) => {
+      const value = e.target.value
+      setValue(value)
+      if (value === 'custom') {
+        setIsCustomAmountInputVisible(true)
+        if (customAmount) {
+          onAmountChanged(customAmount)
+        }
+      } else {
+        setIsCustomAmountInputVisible(false)
+        setPresetAmount(value)
+        onAmountChanged(value)
+      }
+    },
+    [customAmount, onAmountChanged]
+  )
+  const debouncedOnAmountChange = useMemo(
+    () => debounce((amount) => onAmountChanged(amount), INPUT_DEBOUNCE_MS),
+    [onAmountChanged]
+  )
+  useEffect(() => {
+    debouncedOnAmountChange.cancel()
+  }, [debouncedOnAmountChange])
+  const handleCustomAmountChange = useCallback(
+    (amount) => {
+      setCustomAmount(amount)
+      debouncedOnAmountChange(amount)
+    },
+    [setCustomAmount, debouncedOnAmountChange]
+  )
+  return (
+    <>
+      {!isCustomAmountInputVisible ? (
+        <AmountPreview amount={presetAmount} />
+      ) : null}
+      <RadioButtonGroup
+        className={styles.presetAmountButtons}
+        name='AmountPicker'
+        value={value}
+        onChange={handleChange}
+      >
+        {presetAmounts.map((amount) => (
+          <RadioPillButton
+            key={amount}
+            name={'amount'}
+            className={styles.presetAmountButton}
+            label={amount}
+            aria-label={`${amount} audio`}
+            value={amount}
+          />
+        ))}
+        {hideCustomAmount ? null : (
+          <RadioPillButton
+            className={styles.customAmountButton}
+            name={'amount'}
+            label={
+              <span className={styles.customAmountButtonText}>
+                Custom Amount
+              </span>
+            }
+            value={'custom'}
+          />
+        )}
+      </RadioButtonGroup>
+      {isCustomAmountInputVisible ? (
+        <TokenValueInput
+          rightLabelClassName={styles.customAmountLabel}
+          inputClassName={styles.customAmountInput}
+          format={Format.INPUT}
+          placeholder={'Enter an amount'}
+          rightLabel={'$AUDIO'}
+          value={customAmount}
+          isNumeric={true}
```

Comment:
personally like `isNumeric` vs `isNumeric={true}` same is true for strings, `rightLabel='$AUDIO'`, but all good, ideally we have linter autofix everything